fix: resolve Discord webhook 400 error with smart webhook formatting

swarna1101 opened the pull request:
- Add webhook formatter utility that automatically detects webhook types
- Support Discord webhooks with {\"content\": \"message\"} format
- Support Slack webhooks with {\"text\": \"message\"} format  
- Support generic webhooks with raw message content
- Update subscribe command to use webhook formatter for both WebSocket and gRPC
- Add comprehensive tests for webhook formatting functionality
- Update documentation with webhook formatting examples and troubleshooting
- Fix HTTP 400 errors for Discord and Slack webhooks

https://github.com/user-attachments/assets/71148b7b-d8eb-4d8e-a187-09d447ca1b40


https://github.com/user-attachments/assets/60c43fc8-1fda-43ed-8b36-5c2b1f21b87b

hpsing reviewed (changes requested):
I really appreciate your intention, but this has limitations.
Instead, we can let subscribers define their own schema at subscription time. That way, they can decide how the payload should look — either by providing a simple key mapping (e.g. {"body":"message"}) or even a full JSON template (e.g. {"alert":"{{.Message}}", "time":"{{.Timestamp}}"}).

swarna1101 commented:
> I really appreciate your intention, but this has limitations. Instead, we can let subscribers define their own schema at subscription time. That way, they can decide how the payload should look — either by providing a simple key mapping (e.g. {"body":"message"}) or even a full JSON template (e.g. {"alert":"{{.Message}}", "time":"{{.Timestamp}}"}).

fixed it, I removed the formatted I was using. Instead now I am using a template. Full JSON Template with Variables: `./mump2p subscribe --topic=test --webhook="webhook.link" --webhook-schema='{"alert":"{{.Message}}", "time":"{{.Timestamp}}", "topic":"{{.Topic}}"}'`
Would format as: `{"alert": "Hello World", "time": "...", "topic": "test"}`

Tested it with normal webhook, discord and slack, both HTTP and grpc

https://github.com/user-attachments/assets/5e6a28b6-1a8e-41e7-b8fa-16d0d01a15c5


https://github.com/user-attachments/assets/6fa380f0-e212-434f-96dd-9f64bb2130c3


https://github.com/user-attachments/assets/9a558229-bb82-4412-96a5-019131cb5805
